src/test/client/nonblocking.cc

```diff
@@ -712,3 +713,496 @@ TEST_F(TestClient, LlreadvContiguousLlwritevNonContiguous) {
   client->ll_release(fh);
   ASSERT_EQ(0, client->ll_unlink(root, filename, myperm));        
 }
+
+TEST_F(TestClient, LlreadvLlwritevOverlimit) {
+  /* From man page: 
+  POSIX.1  allows  an  implementation to place a limit on the number of
+  items that can be passed in iov.  An implementation can advertise its limit
+  by defining IOV_MAX in <limits.h> or at run time via the return value from
+  sysconf(_SC_IOV_MAX).  On modern Linux systems, the limit is 1024. Back in
+  Linux 2.0 days, this limit was 16 so test if iovcnt exceeds 1024 segments
+  then no data transfer occurs.
+  
+  However going through the libcephfs or the client code, there is no such
+  limit imposed in cephfs, therefore async I/O over the limit should succeed.
+  */
+
+  int mypid = getpid();
+  char filename[256];
+
+  sprintf(filename, "test_llreadvllwritevoverlimitfile%u", mypid);
+
+  Inode *root, *file;
+  root = client->get_root();
+  ASSERT_NE(root, (Inode *)NULL);
+
+  Fh *fh;
+  struct ceph_statx stx;
+
+  ASSERT_EQ(0, client->ll_createx(root, filename, 0666,
+				  O_RDWR | O_CREAT | O_TRUNC,
+				  &file, &fh, &stx, 0, 0, myperm));
+
+  // setup buffer array
+  const int IOV_SEG_OVERLIMIT = 1500;
+  ssize_t bytes_to_write = 0;
+  struct iovec iov_out_overlimit[IOV_SEG_OVERLIMIT];
+  struct iovec iov_in_overlimit[IOV_SEG_OVERLIMIT];
+  for(int i = 0; i < IOV_SEG_OVERLIMIT; ++i) {
+    char out_str[] = "foo";
+
+    // fill iovec structures for write op
+    iov_out_overlimit[i].iov_base = out_str;
+    iov_out_overlimit[i].iov_len = sizeof(out_str);
+    bytes_to_write += iov_out_overlimit[i].iov_len;
+
+    // set up iovec structures for read op
+    char in_str[sizeof(out_str)];
+    iov_in_overlimit[i].iov_base = in_str;
+    iov_in_overlimit[i].iov_len = sizeof(in_str);
+  }
+
+  std::unique_ptr<C_SaferCond> writefinish = nullptr;
+  std::unique_ptr<C_SaferCond> readfinish = nullptr;
+
+  writefinish.reset(new C_SaferCond("test-nonblocking-writefinish-overlimit"));
+  readfinish.reset(new C_SaferCond("test-nonblocking-readfinish-overlimit"));
+
+  int64_t rc;
+  bufferlist bl;
+
+  rc = client->ll_preadv_pwritev(fh, iov_out_overlimit, IOV_SEG_OVERLIMIT,
```

Comment:
Why is this test not using the `write_n_bytes_async` helper introduced in the previous commit?

Reply:
> Why is this test not using the `write_n_bytes_async` helper introduced in the previous commit?

@vshankar the func is more of a filler than a i/o wrapper meaning it is used to fill up fs but won't give control over the iovecs that are needed to compare the content written and read. We can add a func that does it but this doesn't look ugly.

Reply:
The helper name is `write_n_bytes_async` and that to me is a routine that would write `n` bytes from a vector of iovecs. If its a file system filler, perhaps its better to call it that way :)

Reply:
this roots back in `mount.py` where i found a func named `def write_n_mb` which basically does the same thing of writing n bytes to a file ultimately filling up the fs.

Reply:
i can change this to `fill_n_bytes_async`